File: workers/app/models/headline.rb
# app/models/headline.rb
class Headline < ApplicationRecord
  def self.create_from_source(source)
    source.fetch.each.map do |data|
      find_or_create_by(title: data[:title], source: data[:source]) do |headline|
        headline.url = data[:url]
        headline.published_at = Time.current
      end
    end
  end
end

File: backend/Models/Headline.cs
using System;
using System.Collections.Generic;

namespace backend.Models;

public partial class Headline
{
    public long Uid { get; set; }

    public string Title { get; set; } = null!;

    public string Url { get; set; } = null!;

    public string Source { get; set; } = null!;

    public DateTime PublishedAt { get; set; }

    public string Metadata { get; set; } = null!;
    
    // {:summary=>
    //     "\"6 Weeks of Claude Code\" is a blog post that narrates the author's journey through six weeks of learning Claude Code, a programming language. The author shares their personal experiences, challenges, and insights gained throughout this process.",
    //     :keywords=>["Claude Code", "Programming Language", "Learning Journey", "Personal Experience", "Challenges"],
    //     :sentiment=>"Neutral"}

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}
